src/rpc/common/impl/APIVersionParser.h

```diff
@@ -74,6 +78,8 @@ class ProductionAPIVersionParser : public APIVersionParser
 
     util::Expected<uint32_t, std::string>
     parse(boost::json::object const& request) const override;
+
+    friend class RPC::VersionHandler;
```

Comment:
This is probably not the best approach here - `friend` is best used when two or more classes only make sense together as one system. In our case that's not the case so i'd recommend adding getters instead 🥇